lib/actions/user.js

```diff
@@ -134,56 +158,122 @@ export function createOrUpdateUser (userData) {
+      await dispatch(fetchUserMonitoredTrips(accessToken))
+
+      // Finally, navigate to the saved trips page.
+      dispatch(routeTo('/savedtrips'))
+    } else {
+      alert(`An error was encountered:\n${JSON.stringify(message)}`)
+    }
+  }
+}
 
-        // Reload user's monitored trips after add/update.
-        await dispatch(fetchUserMonitoredTrips(accessToken))
+/**
+ * Deletes a logged-in user's monitored trip,
+ * then, if that was successful, refreshes the redux monitoredTrips state.
+ */
+export function deleteUserMonitoredTrip (tripId) {
+  return async function (dispatch, getState) {
+    const { accessToken, apiBaseUrl, apiKey } = getMiddlewareVariables(getState())
+    const requestUrl = `${apiBaseUrl}${API_MONITORTRIP_PATH}/${tripId}`
 
-        // Finally, navigate to the saved trips page.
-        dispatch(routeTo('/savedtrips'))
+    const { message, status } = await secureFetch(requestUrl, accessToken, apiKey, 'DELETE')
+    if (status === 'success') {
+      // Reload user's monitored trips after deletion.
+      await dispatch(fetchUserMonitoredTrips(accessToken))
+    } else {
+      alert(`An error was encountered:\n${JSON.stringify(message)}`)
+    }
+  }
+}
+
+/**
+ * Requests a verification code via SMS for the logged-in user.
+ */
+export function requestPhoneVerificationSms (newPhoneNumber) {
+  return async function (dispatch, getState) {
+    const state = getState()
+    const { number, timestamp } = state.user.lastPhoneSmsRequest
+    const now = new Date()
+
+    // Request a new verification code if we are requesting a different number.
+    // or enough time has ellapsed since the last request (1 minute?).
+    // TODO: Should throttling be handled in the middleware?
+    if (number !== newPhoneNumber || timestamp + 60000 >= now) {
+      const { accessToken, apiBaseUrl, apiKey, loggedInUser } = getMiddlewareVariables(state)
+      const requestUrl = `${apiBaseUrl}${API_OTPUSER_PATH}/${loggedInUser.id}${API_OTPUSER_VERIFYSMS_PATH}/${encodeURIComponent(newPhoneNumber)}`
+
+      const { message, status } = await secureFetch(requestUrl, accessToken, apiKey, 'GET')
+
+      dispatch(setLastPhoneSmsRequest({
+        number: newPhoneNumber,
+        status,
+        timestamp: now
+      }))
+
+      if (status === 'success') {
+        // Refetch user and update application state with new phone number and verification status.
+        // (This also refetches the user's monitored trip, and that's ok.)
+        await dispatch(fetchOrInitializeUser({ accessToken }))
       } else {
-        alert(`An error was encountered:\n${JSON.stringify(result)}`)
+        alert(`An error was encountered:\n${JSON.stringify(message)}`)
       }
+    } else {
+      // Alert user if they have been throttled.
+      // TODO: improve this alert.
+      alert('A verification SMS was sent to the indicated phone number less than a minute ago. Please try again in a few moments.')
```

Comment:
I'm getting this message even when the last code was requested over a minute ago.

Reply:
There was a logic error in the delay check, https://github.com/opentripplanner/otp-react-redux/pull/224/commits/40871d2ad441eabf66785ccb1cd39548b65e7c36 fixes it.

New issue for server-side throttling: https://github.com/ibi-group/otp-middleware/issues/88.

Reply:
I added the expiration text in https://github.com/opentripplanner/otp-react-redux/pull/224/commits/738e9cf9cb5e8dcc09baf147f4249a49046b562f.